GH-672: review 7

bertllll opened the pull request.

Commit:
GH-672: ending a big rearrangement; stable with all the tests again

Commit:
GH-672: more refactoring

Commit:
GH-672: missing import

dnwiebe commented on node/src/blockchain/blockchain_interface/blockchain_interface_null/mod.rs:
```diff
@@ -182,15 +184,20 @@ mod tests {
-        let result = make_subject(test_name)
-            .build_blockchain_agent(&wallet, &persistent_config)
-            .unwrap();
+        let result = make_subject(test_name).build_blockchain_agent(&wallet, &persistent_config);
```
You thought maybe you should have `subject` as its own variable. I have an opinion, but I'm leaving it up to you.

dnwiebe commented on node/src/blockchain/blockchain_interface/data_structures/errors.rs:
```diff
@@ -0,0 +1,259 @@
+pub enum PayableTransactionError {
+    MissingConsumingWallet,
+    GasPriceQueryFailed(String),
+    TransactionCount(BlockchainError),
```
There might be a better name for this.

dnwiebe commented on node/src/blockchain/blockchain_interface/data_structures/errors.rs:
```diff
@@ -0,0 +1,259 @@
+                write!(f, "Unsuccessful gas price query: \"{}\"", msg)
+            }
+            Self::TransactionCount(blockchain_err) => {
+                write!(f, "Transaction count fetching failed: {}", blockchain_err)
```
Don't forget to change this string as well

dnwiebe commented on node/src/blockchain/blockchain_interface/data_structures/errors.rs:
```diff
@@ -0,0 +1,259 @@
+pub enum BlockchainAgentBuildError {
+    GasPrice(PersistentConfigError),
+    TransactionFeeBalance(Wallet, BlockchainError),
+    MasqBalance(Wallet, BlockchainError),
```
If we ever get this working on other blockchains, the token there is probably not going to be called MASQ. (Or maybe it is; who knows?) Anyway, there may come a time when we'll need this name to be more general. Is this that time? Maybe not. Okay, never mind--unless you want to.

Commit:
GH-672: review 7 answered

masqrauder committed:
GH-600: Add get_block_number (#193)

* Squashing:
GH-600: Add get_block_number
GH-600: Address PR feedback
GH-585: Experiment with adapting test for platform specific error codes
GH-585: Provide Windows platform specific error code and message
GH-600: Rebase master into branch

* GH-600: Handle expected batch, got single response

* Resolve rebase issues

* GH-600: Adapt to new Pokt message changes

* Use start_block in the event of an error

* GH-600: Return blockchain scan error to update max_block_count

* GH-600: Resolve blockchain scanning

* Starts Accountant
* Adds Receivables scan timeout (may become a configuration parameter in another card)
* Stores `max_block_count` in the config table when processing error messages providing a limit
* Adds Max Block Count to configuration response
* Increments CURRENT_SCHEMA_VERSION
* Adds schema migration for `max_block_count`
* Improves blockchain scan error handling and logging

* GH-600: Add test for scan time out

* GH-600: Revert scanner changes

* GH-600: Address review feedback

* GH-600: Address review 6 feedback

* GH-600: Use From traits to be concise

Commit:
GH-672: merged master in

Commit:
GH-672: removed a not suitable comment

Commit:
GH-672: review 8 answered

Commit:
GH-672: lower interface renames
